src/notifications/discord.rs

```diff
@@ -0,0 +1,52 @@
+use crate::notifications::NotificationMessage;
+use inflections::case::to_title_case;
+use log::debug;
+use serde::{Deserialize, Serialize};
+use std::env;
+
+const SUCCESS_COLOR: i32 = 4961603;
+const FAILURE_COLOR: i32 = 16388413;
+const GENERIC_COLOR: i32 = 32614;
+const DISCORD_WEBHOOK_BASE: &str = "https://discord.com/api/webhooks";
+
+pub fn is_discord_webhook(webhook_url: &str) -> bool {
+  webhook_url.starts_with(DISCORD_WEBHOOK_BASE)
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct DiscordWebHookEmbed {
+  title: String,
+  description: String,
+  color: i32,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct DiscordWebHookBody {
+  content: String,
+  embeds: Vec<DiscordWebHookEmbed>,
+}
+
+pub fn build_discord_payload(event: NotificationMessage) -> DiscordWebHookBody {
+  let server_name = env::var("NAME").unwrap_or_else(|_| String::from("Your Valheim Server"));
+  let name = event.event_type.to_string();
+  let color = if name.contains("Successful") {
+    SUCCESS_COLOR
+  } else if name.contains("Failure") {
+    FAILURE_COLOR
+  } else {
+    GENERIC_COLOR
+  };
```

Comment:
You could actually encode this information with the type system instead of relying on matching information within strings. Something like

```rust
enum Color {
    Success = 0x4B_B5_43,
    Failure = 0xFA_11_3D,
    Generic = 0x00_7F_66,
}

impl From<NotificationEventType> for Color {
    fn from(event: NotificationEventType) -> Self {
        use NotificationEventType::{
            UpdateSuccessful,
            UpdateFailed,
            StartSuccessful,
            StartFailed,
            StopSuccessful,
            StopFailed,
        };

        match event {
            UpdateSuccessful | StartSuccessful | StopSuccessful => Self::Success,
            UpdateFailed | StartFailed | StopFailed => Self::Failure,
            _ => Self::Generic,
        }
    }
}
```

and from there you can just `let color = Color::from(event.event_type) as i32;` to get the value.

Reply:
What is the best way to convert 0x4B_B5_43 to the u32 variant? Discord requires a number to be passed in for the message color.

Reply:
You can just do `Color::Success as u32` to get it as a `u32`

Reply:
Rust by example has a [nice snippet](https://doc.rust-lang.org/rust-by-example/custom_types/enum/c_like.html) on it